src/h2olog/h2olog.cc

```diff
@@ -230,6 +252,13 @@ int main(int argc, char **argv)
         exit(EXIT_FAILURE);
     }
 
+    if (list_and_exit) {
+        for (const auto &usdt : tracer->usdt_probes()) {
+            printf("%s:%s\n", usdt.provider.c_str(), usdt.name.c_str());
+        }
+        exit(EXIT_SUCCESS);
+    }
```

Comment:
I think there is ambiguity in what `-l` means.

If it is a list of probes that are _selected,_ then the code should go through `event_type_filters`.

If it is a list of probes that are _available,_ then the code should go through `available_usdts`. Also, the logic should be moved to right after `case 'l':` and `list_and_exit` can be removed.